### PR TITLE
Goodfet update

### PR DESCRIPTION
**Pull request by taylorcenters**
re-implemented goodfet changes from inode's PR #216 to work with python3

### CODE REVIEW

**Review by rmspeers — APPROVED** [review]

**Comment by phipolis** [comment]
Thanks so much for this!! Is the [firmware/apimotev4_202011.hex file](firmware/apimotev4_202011.hex) included in #216 needed?

**Comment by taylorcenters** [comment]
@phipolis hi! Not a problem - I think that new firmware will need to be loaded on the apimote to correspond to the goodfet changes. I'm trying to think of a better way in the future to distribute built firmware - I might try and add a github pipeline to build and publish the firmware artifact - I feel a bit uneasy about just committing pre-built firmware hex files.

File: killerbee/GoodFETCCSPI.py [hunk 1]
```diff
@@ -77,12 +78,12 @@ def peek(self,reg,bytes=2):
...
         
         self.writecmd(self.CCSPIAPP,0x02,len(data),data);
-        try:
-            toret = (self.data[2] + (self.data[1] << 8))
-        except Exception as e:
-            print("issue in peeking for a register")
-            print(e)
-            toret = self.data[1] << 8
+  
+        toret=(
```
Comment (taylorcenters) [inline killerbee/GoodFETCCSPI.py:82]: These ord's are causing errors

```
======================================================================                                
ERROR: test_init (test_apimote_driver.TestApimoteDriver)                                              
----------------------------------------------------------------------                                
Traceback (most recent call last):                 
  File "/Users/taylor/workspace/killerbee/tests/test_apimote_driver.py", line 11, in test_init        
    driver=APIMOTE(os.environ['APIMOTE_DEVSTRING'])                                                   
  File "/Users/taylor/.pyenv/versions/killerbee/lib/python3.9/site-packages/killerbee-3.0.0b1-py3.9-macosx-10.15-x86_64.egg/killerbee/dev_apimote.py", line 53, in __init__
    self.handle.setup()                            
  File "/Users/taylor/.pyenv/versions/killerbee/lib/python3.9/site-packages/killerbee-3.0.0b1-py3.9-macosx-10.15-x86_64.egg/killerbee/GoodFETCCSPI.py", line 25, in setup
    self.poke(0x11, 0x0AC2 & (~0x0800)); #MDMCTRL0, promiscuous                                       
  File "/Users/taylor/.pyenv/versions/killerbee/lib/python3.9/site-packages/killerbee-3.0.0b1-py3.9-macosx-10.15-x86_64.egg/killerbee/GoodFETCCSPI.py", line 92, in poke
    if self.peek(reg,bytes)!=val and reg!=0x18:    
  File "/Users/taylor/.pyenv/versions/killerbee/lib/python3.9/site-packages/killerbee-3.0.0b1-py3.9-macosx-10.15-x86_64.egg/killerbee/GoodFETCCSPI.py", line 83, in peek
    ord(self.data[2])+                             
TypeError: ord() expected string of length 1, but int found     
```

**Comment by phipolis** [comment]
> @phipolis hi! Not a problem - I think that new firmware will need to be loaded on the apimote to correspond to the goodfet changes. I'm trying to think of a better way in the future to distribute built firmware - I might try and add a github pipeline to build and publish the firmware artifact - I feel a bit uneasy about just committing pre-built firmware hex files.

@taylorcenters I'm glad you're thinking about that, I was unclear about the hex files significance and couldn't find any instructions on their use with apimote.

**Pull request merged**